File: tasks/2.1/2.1.ts
function binraySearch(arr: number[], element: number): number {
    let l = 0
    let r = arr.length - 1

    while (l <= r) {
        const middle = (l + r) / 2
        if (arr[middle] > element) {
            r = middle - 1
        } else if (arr[middle] < element) {
            l = middle + 1
        } else {
            return middle
        }
    }
    return l
}

File: tasks/2.1/2.1.py
def binary_search(arr, element):
    l = 0
    r = len(arr) - 1
    if (arr[r] < element):
        return r + 1
    if (arr[l] > element):
        return 0
    while (l <= r):
        middle = (l + r) // 2
        if (arr[middle] > element):
            r = middle - 1
        elif (arr[middle] < element):
            l = middle + 1
        else: 
            return middle
    return l

n = int(input())
arr = list(map(int, input().split()))
element = int(input())
print(binary_search(arr, element))